### DISCUSSION TITLE
How to compile and other questions

### DISCUSSION THREAD

**Discussion by peterphonic** [uNetworking/uWebSockets#1535] [General]
That is a silly question, but I searched everywhere and I can't find instructions to compile uWebSocket.  Are there any?  I can see Makefile and build.c, but I don't really know how to do.

I would like to compile on windows 10, x64, vs2017 with openSSL.

Regarding OpenSSL, I saw on build.c that you strongly recommend boringSSL instead.  
1. Is it easier to use boringSSL
2. I quickly read on internet that BoringSSL is not intended for general use and the programming interface is not stable.  SHould I use boringSSL
3. I already created pem, cert and key with openSSL.  Are those files will work with boringSSL?

Thank you very much

**Comment by uNetworkingAB** [comment]
If you just want to play around you can get it from vcpkg that Microsoft provide. Read about vcpkg, it should be something like `vcpkg install uwebsockets:x64-windows` or something like that.

Windows is heavily discouraged in general, consider it a second class platform. So giving per individual instructions for how you as individual can get your set up working is out of scope for this project. On Linux, it's just "make" and you're there in seconds.

**Reply by peterphonic** [reply]
Thank you for your quick reply.  I agree that Windows is a second class platform.  We need to add a  websocket client with SSL into our existing windows program.  Unfortunately, I have to stick with this platform :(.    Anyway thank you, and I have to admit that your client and server interface is really elegant compared to many other websocket projects....
